src/components/Badge/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { ComponentProps, ReactElement, ReactNode,forwardRef } from "react";
+import clsx from "clsx";
+import { TGenericSizes } from "../../types";
+import { Text } from "../Text";
+import "./Badge.scss";
+
+type TVariant = "contained" | "bordered";
+type TColor =
+    | "blue"
+    | "light-blue"
+    | "general"
+    | "purple"
+    | "success"
+    | "warning"
+    | "danger"
+    | "gold"
+    | "green"
+type TPadding = "tight" | "loose";
+
+interface BadgeProps extends ComponentProps<"div"> {
+    children?: ReactNode;
```

Comment:
`ReactNode` already includes `undefined` @yaswanth-deriv.
```suggestion
    children: ReactNode;
```